api/middlewares/auth-middleware.go

```diff
@@ -0,0 +1,120 @@
+package middlewares
+
+import (
+	"encoding/json"
+	"errors"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	jwtmiddleware "github.com/auth0/go-jwt-middleware"
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/gin-gonic/gin"
+)
+
+// JSONWebKeys struct
+type JSONWebKeys struct {
+	Kty string   `json:"kty"`
+	Kid string   `json:"kid"`
```

Comment:
these short keywords are common in go? I would normally prefer full words.

Reply:
Yes, these words are too short, I too prefer a little description there in naming. But these are coming from JSON web token RFC. It helps to map with RFC model.

More details here https://tools.ietf.org/html/rfc7517. 

The RFC is kind of one time thing, and won't be that frequent. 
We will use full words for the api implementations.